### PR TITLE
Include exclude projects enhancement

### PR DESCRIPTION
**Pull request by ivan-danilov**

### CODE REVIEW

**Comment by ivan-danilov** [comment]
Fixes #75

File: src/GitLink/Helpers/ProjectHelper.cs [hunk 1]
```diff
@@ -116,5 +117,37 @@ public static Project LoadProject(string projectFile, string configurationName,
...
+        public static bool ShouldBeIgnored(string projectName, ICollection<string> projectsToInclude, ICollection<string> projectsToIgnore)
+        {
+            Argument.IsNotNull(() => projectName);
+
+            if (projectsToIgnore.Any(projectToIgnore => ProjectNameMatchesPattern(projectName, projectToIgnore)))
+            {
+                return true;
+            }
+
+            if (projectsToInclude.Count == 0) return false;
```
Comment (GeertvanHorrik) [inline src/GitLink/Helpers/ProjectHelper.cs]: Just our coding standards, use return false inside { and }

File: src/GitLink/Helpers/ProjectHelper.cs [hunk 2]
```diff
@@ -116,5 +117,37 @@ public static Project LoadProject(string projectFile, string configurationName,
...
+
+        // pattern may be either a literal string, and then we'll be comparing literally ignoring case
+        // or it can be a regex eclosed in slashes like /this-is-my-regex/
+        private static bool ProjectNameMatchesPattern(string projectName, string pattern)
+        {
+            if (pattern.Length > 2 && pattern.StartsWith("/") && pattern.EndsWith("/"))
+            {
+                var ignoreRegex = new Regex(pattern.Substring(1, pattern.Length - 2), RegexOptions.IgnoreCase);
+                if (ignoreRegex.IsMatch(projectName))
+                    return true;
```
Comment (GeertvanHorrik) [inline src/GitLink/Helpers/ProjectHelper.cs]: Encapsulate with { and }

File: src/GitLink/Helpers/ProjectHelper.cs [hunk 3]
```diff
@@ -116,5 +117,37 @@ public static Project LoadProject(string projectFile, string configurationName,
...
+                return true;
+            }
+
+            return false;
+        }
+
+        // pattern may be either a literal string, and then we'll be comparing literally ignoring case
+        // or it can be a regex eclosed in slashes like /this-is-my-regex/
+        private static bool ProjectNameMatchesPattern(string projectName, string pattern)
+        {
```
Comment (GeertvanHorrik) [inline src/GitLink/Helpers/ProjectHelper.cs]: Add Argument.IsNotNull(() => pattern)

**Pull request merged**